### PR TITLE
Fix some potential bugs and add a masked API key property

### PR DESCRIPTION
**Pull request by raman325**
Potential bugs fixed:
- force no compression when sending data since the server occasionally seems to respond with a 400 `Can not decode content-encoding: gzip`
- Switch from json.dumps to using aiohttp's internal serialization logic
- Remove filter for content-type check on response - we will handle a parsing exception anyway

Feature:
- Add masked API key property. We want to mask the key in the coordinator name so logs can more easily be posted

### CODE REVIEW

File: pytomorrowio/const.py [hunk 1]
```diff
@@ -6,8 +6,8 @@
 
 MAX_FIELDS_PER_REQUEST = 55
 
-HEADER_DAILY_API_LIMIT = "X-RateLimit-Limit-Day"
-HEADER_REMAINING_CALLS_IN_SECOND = "X-RateLimit-Remaining-Second"
+HEADER_DAILY_API_LIMIT = "X-RateLimit-Limit-day"
+HEADER_REMAINING_CALLS_IN_SECOND = "X-RateLimit-Remaining-second"
```
Comment (lymanepp) [inline pytomorrowio/const.py:10]: HTTP headers names should be handled case-insensitively.

File: pytomorrowio/pytomorrowio.py [hunk 2]
```diff
@@ -182,34 +187,38 @@ def available_fields(
...
+                compress=False,
             )
-            resp_json = await resp.json()
+            resp_json = await resp.json(content_type=None)
         except ClientConnectionError as error:
             raise CantConnectException() from error
 
         self._rate_limits = CIMultiDict(
-            {k: int(v) for k, v in resp.headers.items() if "ratelimit" in k.lower()}
+            {k: int(v) for k, v in resp.headers.items() if "limit" in k.lower()}
```
Comment (lymanepp) [inline pytomorrowio/pytomorrowio.py:221]: Is this change necessary?

Comment (raman325) [reply pytomorrowio/pytomorrowio.py:221]: no, this was another thing I was changing while testing the issues reported in https://github.com/home-assistant/core/issues/77355

File: pytomorrowio/const.py (hunk 1 above)
Comment (raman325) [reply pytomorrowio/const.py]: I have a support ticket open about them not sending the headers, which are still documented in their docs

Comment (lymanepp) [reply pytomorrowio/const.py]: After further review, since the code is using [CIMultiDict](https://multidict.readthedocs.io/en/stable/) it is already case-insensitive. Changing the case of these should be irrelevant.

Comment (raman325) [reply pytomorrowio/const.py]: yeah - reverted it but it was a noop change. The real problem is that they aren't sending the right headers
